### DISCUSSION TITLE
Roborock token support (with code)

### DISCUSSION THREAD

**Discussion by NewsGuyTor** [rytilahti/python-miio#1572] [General]
Now that grabbing your Roborock token has been made pretty easy with this code,
https://forum.iobroker.net/topic/56098/tester-gesucht-roborock-api

https://gist.github.com/rovo89/dff47ed19fca0dfdda77503e66c2b7c7

I think that support for the Roborock token should be implemented. That way you can use both the Roborock app (way better than Mi Home) and Home Assistant at the same time.

**Comment by bobzer** [comment]
I don't know how to implement this but I agree that would really cool

**Reply by 4c0d3r** [reply]
Script is broke, the returned data is null.

**Reply by 4c0d3r** [reply]
Just attempted to use the referenced script above, trying to get info back to the authors that they are currently broken.

**Reply by humbertogontijo** [reply]
The script works for me. Just had to change the url from https://euiot.roborock.com to https://usiot.roborock.com

**Reply by 4c0d3r** [reply]
I had already updated the url, but had to delete the 'userdata.json' from disk before it would work.   Thank you for letting me know it worked for you.

**Reply by humbertogontijo** [reply]
I've created a python script based on those js files if it helps https://github.com/humbertogontijo/homeassistant-roborock

**Reply by andrediaslopes** [reply]
> I've created a python script based on those js files if it helps https://github.com/humbertogontijo/homeassistant-roborock

Thanks for that! I was waiting a lot of time for this integration. I really like the roborock app and wanted to use it with home assistant as well.

**Comment by Frederick-G764** [comment]
I receive the error: 
Cannot set properties
TypeError: Cannot set properties of undefined (setting 'Authorization')

**Reply by andrediaslopes** [reply]
I was also receiving this error, but with the Python script of @humbertogontijo (https://github.com/rytilahti/python-miio/discussions/1572#discussioncomment-4339160) it worked like a charm!
